firefox-ios/Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -43,6 +43,26 @@ extension WKWebView {
         let safeUrl = url.absoluteString.replacingOccurrences(of: "'", with: apostropheEncoded)
         evaluateJavascriptInDefaultContentWorld("location.replace('\(safeUrl)')")
     }
+
+    /// - Parameters:
+    ///     - script: String representing javascript to be evaluated
+    ///     - arguments: A dictionary of the arguments to pass to the function call
+    ///     - completion: Result containing any? data and an error
+    public func callAsyncJavaScriptInDefaultContentWorld(_ script: String,
+                                                         arguments: [String: Any],
+                                                         completion: @escaping (Result<Any?, Error>) -> Void ) {
+        self.callAsyncJavaScript(script,
+                                 arguments: arguments,
+                                 in: nil,
+                                 in: .defaultClient) { result in
+            switch result {
+            case .success(let value):
+                completion(.success(value))
+            case .failure(let error):
+                completion(.failure(error))
```

Comment:
nit: log for failure to capture the error

Reply:
No logs required here.